[cleanup crusade] Lint bash scripts on letter A

paulfantom opened the pull request:
Running shellcheck on scripts from `charts.d` and linting them.

This PR disables shebang checking in all files and lints scripts starting with letter A. More linting will follow in another PRs so we don't have one giant PR with everything and this will be easier to review and fix.

ktsaou commented on charts.d/apache.chart.sh:
```diff
@@ -98,16 +98,17 @@ apache_detect() {
 	ret=$?
 	IFS="${oIFS}"
 
-	[ $ret -ne 0 -o "${#apache_response[@]}" -eq 0 ] && return 1
```
Why is this needed? Readability is killed with this.

paulfantom replied:
I would argue that having multiple operators starting with `-` kills readability. Using `[ ]` neatly groups conditions and `&&` or `||` makes it clear what is a condition and how do they apply.

And as with every shellcheck code, there is an explanation: https://github.com/koalaman/shellcheck/wiki/SC2166

ktsaou replied:
ok

ktsaou commented:
So, shellcheck does like binary operators in `[` ?
Why?

paulfantom commented:
SC2166 explains it: https://github.com/koalaman/shellcheck/wiki/SC2166

ktsaou commented:
Wow! This means a lot of work (installer, makeself, charts.d.plugin, alarm-notify.sh)

paulfantom commented:
Yeah, that's why my PRs are called "crusade" :stuck_out_tongue:

paulfantom commented:
I am also considering using:
```sh
#! /bin/echo file should be sourced
```
as a shebang for bash "lib" files instead of disabling shebang shellcheck. @ktsaou what do you think?

ktsaou commented on charts.d/apcupsd.chart.sh:
```diff
@@ -43,11 +44,12 @@ apcupsd_check() {
             error "cannot get information for apcupsd server ${host} on ${apcupsd_sources[${host}]}."
             failed=$((failed + 1))
-        elif [ $(apcupsd_get "${apcupsd_sources[${host}]}" | awk '/^STATUS.*/{ print $3 }') != "ONLINE" ]
+        elif [ "$(apcupsd_get \"${apcupsd_sources[${host}]}\" | awk '/^STATUS.*/{ print $3 }')" != "ONLINE" ]
```
are you sure backslases are needed here? They should not, since they are inside `$()`.

ktsaou commented:
check the last comment, about backslashes. Other than that, I think we can merge this.

paulfantom commented:
Now it should be ok. Fixed backslashes and I added something that simulates shebang in shellcheck without adding shebang: https://github.com/koalaman/shellcheck/wiki/Directive#shell

Merge?

ktsaou commented:
yeap! let's wait travis...

ktsaou commented:
ok.

I think I saw on travis that on my fork, it also builds "Docker" and "Github". Should these be run only on `firehol/netdata`?

paulfantom commented:
Doesn't matter as long as secrets are only pinned to `firehol/netdata`.

ktsaou commented:
It takes time... (and travis resources) ...

paulfantom commented:
True, but I cannot do anything about it thou.

ktsaou commented:
ok

ktsaou commented:
So, you mean you can't check the repo the commits are coming from?

paulfantom commented:
I am checking only branch name, didn't see anything for checking repo in new travis build stages.

paulfantom commented:
You can always disable travis for your fork :)

ktsaou commented:
No, it is not about me. It is about the PRs...

paulfantom commented:
I don't understand. PRs are not building "release" stage.

ktsaou commented:
ok. So you are saying that PRs do not build "Docker" and "Github". I don't recall exactly when I saw it. Probably it was not PR... I don't know.

Anyway we will see later.

btw, you could check of the token variables. If they don't exist or are empty, don't build these.

paulfantom commented:
> btw, you could check of the token variables. If they don't exist or are empty, don't build these.

What for? "release" should happen only from master branch and travis already has mechanism for that. 

https://github.com/firehol/netdata/blob/5c1a47c9366401c2bddecbb3425eec46400ae860/.travis.yml#L15

ktsaou commented:
ok... thanks!
